src/client.rs

```diff
@@ -207,7 +207,15 @@ impl Client {
     /// Do initial sync and return rooms and new invites. Then register event handler for future syncs and notify events.
     pub async fn sync(&self, user_id: &UserId) -> SyncResult {
         let client = self.inner.lock().await;
+        let now = Instant::now();
         let response = client.sync_once(SyncSettings::default()).await;
+        self.event_notifier
+            .send(Event::RequestDuration((
+                UserRequest::InitialSync,
```

Comment:
Is this event particularly interesting to us?

Reply:
I think it is, a slow initial sync would say those workers are not responding as expected.